src/SwiftOutputChannel.ts

```diff
@@ -0,0 +1,50 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2021 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from 'vscode';
+import { Disposable } from 'vscode-jsonrpc';
+
+export class SwiftOutputChannel implements Disposable {
+    private channel: vscode.OutputChannel;
+
+    constructor() {
+        this.channel = vscode.window.createOutputChannel("Swift");
+    }
+
+    dispose() {
+        this.channel.dispose();
+    }
+
+    log(message: string, folder?: string) {
+        if (folder !== undefined) {
+            folder += ": ";
+        }
+        this.channel.appendLine(`${this.nowFormatted}: ${folder ?? ""}${message}`);
+    }
+
+    logStart(message: string, folder?: string) {
```

Comment:
Can we automate part of this start/end combination? Perhaps adding the triple dots automatically, and letting `logEnd` print either an optional error parameter, or a default "Done" message for consistency?

Reply:
I wanted to leave some flexibility here, don't necessarily want to always post `done`

Reply:
OK